### ISSUE TITLE
1.16.5 - 36.2.2 - 0.15.2 - Server "Failed to store chunk"

### ISSUE THREAD

**Issue by Snaitf** [MehVahdJukaar/Supplementaries#95]
Minecraft: 1.16.5
Forge: 36.2.2
Supplementaries: 0.15.2
Selene: 1.6

Server will not save any chunks when this mod is loaded.  Tested with Supplementaries and Selene only.

Debug Log:
https://hastebin.com/ahutigowul.yaml

**Comment by asiekierka** [comment]
The issue is that the structure `net.mehvahdjukaar.supplementaries.world.structures.WaySignStructure`'s feature name getter returns null - that implies it's not properly registered with Forge.

Duplicate of https://github.com/MehVahdJukaar/Supplementaries/issues/93

**Comment by DPurple** [comment]
Well , in net.minecraft.world.gen.feature.structure.Structure.java line 235
getFeatureName() is using STRUCTURES_REGISTRY which only touched by vanilla structures.
DeferredRegister one only simple use setRegistryName for forge itself but didn't touch STRUCTURES_REGISTRY 
so calling super.getFeatureName() will result null because it doesn't exists in STRUCTURES_REGISTRY.
I think this should be a bug of forge, the solution is using forge registry in WaySignStructure.getFeatureName() instead of super.getFeatureName()

**Comment by DPurple** [comment]
I think this may fix the problem but ugly...
```
    @Override
    public String getFeatureName() {
        retrun net.minecraftforge.registries.ForgeRegistries.STRUCTURE_FEATURES.getKey(this).toString();
    }
```

**Comment by DPurple** [comment]
Sorry I think simple patch getFeatureName() is not a good idea which ChunkSerializer tried to use STRUCTURES_REGISTRY too, and locate command.
Have correct STRUCTURES_REGISTRY should be important...

**Comment by MehVahdJukaar** [comment]
Fixed in 0.15.2b
